content/docs/reference/tutorials/aws/tutorial-ec2-webserver.md

```diff
@@ -44,10 +76,58 @@ In this tutorial, we'll use JavaScript to deploy a simple webserver EC2 instance
     exports.publicHostName = server.publicDns;
     ```
 
-    This example uses the [@pulumi/aws] package to create and manage two AWS resources: an
-    [aws.ec2.SecurityGroup][Security Group], which allows access for incoming SSH access, and an
-    [aws.ec2.Instance][EC2 Instance], which is created in that security group using the appropriate Amazon
-    Machine Image (AMI) for the region where you deploy the program.
+    ```typescript
+    const aws = require("@pulumi/aws");
+
+    const size = "t2.micro";     // t2.micro is available in the AWS free tier
+    const ami  = "ami-0ff8a91507f77f867"; // AMI for Amazon Linux in us-east-1 (Virginia)
+
+    const group = new aws.ec2.SecurityGroup("webserver-secgrp", {
+        ingress: [
+            { protocol: "tcp", fromPort: 22, toPort: 22, cidrBlocks: ["0.0.0.0/0"] },
+        ],
+    });
+
+    const server = new aws.ec2.Instance("webserver-www", {
+        instanceType: size,
+        securityGroups: [ group.name ], // reference the security group resource above
+        ami: ami,
+    });
+
+    exports.publicIp = server.publicIp;
+    exports.publicHostName = server.publicDns;
+    ```
+
+    ```python
+    import pulumi
+    from pulumi_aws import ec2
+    from ami import get_linux_ami
+
+    size = 't2.micro'
+
+    group = ec2.SecurityGroup('webserver-secgrp',
+        description='Enable HTTP access',
+        ingress=[
+            { 'protocol': 'tcp', 'from_port': 22, 'to_port': 22, 'cidr_blocks': ['0.0.0.0/0'] }
+        ])
+
+    server = ec2.Instance('webserver-www',
+        instance_type=size,
+        security_groups=[group.name], # reference security group from above
+        ami=get_linux_ami(size))
+
+    pulumi.export('publicIp', server.public_ip)
+    pulumi.export('publicHostName', server.public_dns)
+    ```
+
+    This example uses the [@pulumi/aws] package in JavaScript and TypeScript code and the [pulumi_aws](/docs/reference/pkg/python/pulumi_aws) package in Python code to create two resources:
+
+    * **Security Group**, which allows access for incoming SSH access
+        * JavaScript and TypeScript: [aws.ec2.SecurityGroup][Security Group]
+        * Python: [ec2.SecurityGroup](/docs/reference/pkg/python/pulumi_aws/ec2#pulumi_aws.ec2.SecurityGroup)
```

Comment:
I generally like to use `relref` for internal linking, because Hugo will ensure the page exists, and break the build otherwise. This is helpful when moving content around on the site.

```suggestion
        * Python: [ec2.SecurityGroup]({{< relref "/docs/reference/pkg/python/pulumi_aws/ec2#pulumi_aws.ec2.SecurityGroup" >}})
```


Alternatively, you could also use reference style links, since that's what's being used for the JS/TS link, e.g. at the bottom of the page add something like:

```
[Python Security Group]: {{< relref "/docs/reference/pkg/python/pulumi_aws/ec2#pulumi_aws.ec2.SecurityGroup" >}}
```

And then change the bullet here to be:

```
* Python: [ec2.SecurityGroup][Python Security Group]
```